sql/src/test/java/org/apache/druid/sql/calcite/QueryTestRunner.java

```diff
@@ -122,15 +123,15 @@ public static class QueryResults
     public final List<Query<?>> recordedQueries;
     public final Set<ResourceAction> resourceActions;
     public final RuntimeException exception;
-    public final PlannerCaptureHook capture;
+    public final PlannerHook capture;
 
     public QueryResults(
         final Map<String, Object> queryContext,
         final String vectorizeOption,
         final RelDataType sqlSignature,
         final List<Object[]> results,
         final List<Query<?>> recordedQueries,
-        final PlannerCaptureHook capture
+        final PlannerHook capture
```

Comment:
is there a reason to use `PlannerHook` instead of `PlannerCaptureHook` here?

Reply:
I changed it to have the interface instead ; so that later other implementations can be used

...however it can be interpreted like the `system captures` a set of things for you and here they are in the `QueryResults`

I would rather see the results of this hook like a map of values or something instead of the hook itself...
I find this whole thing oddly designed as the interface has only `captureX` methods ; meanwhile the `PlannerCaptureHook` has the `getX`-s

should I change it back?

Reply:
Yes, I agree that the interface is a bit odd in terms of using it practically. But given that we specifically know how we want to use it in the tests, I think we can use `PlannerCaptureHook` itself - since that provides the interface + extra capabilities/bridge methods that we need.